### PR TITLE
Add handler earlier

### PR DESCRIPTION
**Pull request by Macroz**
I noticed that the `:handler` role is added only from the applications so there needs to be an application before a person is a handler.

# Checklist for author

Remove items that aren't applicable, check items that are done.

## Reviewability
- [ ] Link to issue

## Documentation
- [x] Update changelog if necessary

## Testing
- [x] Complex logic is unit tested
- [x] Valuable features are integration / browser / acceptance tested automatically

## Follow-up
- [ ] New tasks are created for pending or remaining tasks

### CODE REVIEW

**Review by opqdonut — APPROVED** [review]

File: src/clj/rems/middleware.clj [hunk 1]
```diff
@@ -73,6 +74,7 @@
                                (when context/*user*
                                  (set/union (roles/get-roles (getx-user-id))
                                             (organizations/get-all-organization-roles (getx-user-id))
+                                            (workflows/get-all-workflow-roles (getx-user-id))
                                             (applications/get-all-application-roles (getx-user-id))))
```
Comment (opqdonut) [inline src/clj/rems/middleware.clj:78]: some redundancy here but I guess it won't matter, and we need to track the per-application-roles anyway for application permissions

File: src/clj/rems/db/workflow.clj [hunk 2]
```diff
@@ -52,3 +52,8 @@
 
 (defn join-workflow-licenses [workflow]
   (assoc workflow :licenses (get-workflow-licenses (:id workflow))))
+
+(defn get-all-workflow-roles [userid]
+  (when (some #(contains? (set (map :userid (get-in % [:workflow :handlers]))) userid)
+              (get-workflows nil))
```
Comment (opqdonut) [inline src/clj/rems/db/workflow.clj:58]: I wonder if we need to cache this? It does get run on every request, and there can be quite a few workflows. Oh well, I guess time will tell.

File: src/clj/rems/middleware.clj (hunk 1 above)
Comment (Macroz) [reply src/clj/rems/middleware.clj:78]: Yes I think that's fine.

File: src/clj/rems/db/workflow.clj (hunk 2 above)
Comment (Macroz) [reply src/clj/rems/db/workflow.clj:58]: I added a tech debt task about this concern https://github.com/CSCfi/rems/issues/2647

**Pull request merged**